packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -542,6 +547,8 @@ class IsolateManager extends Disposer {
   ValueListenable<IsolateRef> get mainIsolate => _mainIsolate;
   final _mainIsolate = ValueNotifier<IsolateRef>(null);
 
+  final _isolatesCache = <String, Isolate>{};
```

Comment:
there is already an isolateCache. Use it directly for cases where a cache is suitable rather than creating another one. I believe it is in IsolateManager.

Reply:
It looks like the existing cache is being used to cache the isolate states, not the isolate itself. It never actually sends a request for the isolate[1], which we need to do. Though I don't understand how the existing cache is getting the isolate without requesting it explicitly, so I might be mistaken!

https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/service_manager.dart#L734-L738

Reply:
Realized why the pre-existing cache wasn't working is because I was using it in `_loadIsolateState`. Now we are using the cache everywhere except `loadIsolateState`, which is where we make the request for the isolate.